[GLib] Update the libsysprof-capture source tree to version 48.1

psaavedra opened the pull request:
#### 328caa207560c6734aeb263aadab46ddfef47956
<pre>
[GLib] Update the libsysprof-capture source tree to version 48.1
<a href="https://bugs.webkit.org/show_bug.cgi?id=298313">https://bugs.webkit.org/show_bug.cgi?id=298313</a>

Reviewed by Adrian Perez de Castro.

Includes the following upstream changes:

git log --pretty=oneline --abbrev-commit 47.0...48.1 -- src/libsysprof-capture

c7401534 libsysprof-capture: Clean ring buffer memory for counters
b129c7de mapped-ring-buffer: Round to the shared memory boundary on hppa
5151345e libsysprof-capture: fix license of mapped-ring-buffer
45b3f99e libsysprof-capture: unsigned cast before shift
d3058129 libsysprof-capture: Set initial counter id to pid
788b1995 libsysprof-capture: add version macros for 48

* Source/ThirdParty/libsysprof-capture/README.WebKit:
* Source/ThirdParty/libsysprof-capture/mapped-ring-buffer.c:
(mapped_ring_buffer_new_reader):
* Source/ThirdParty/libsysprof-capture/sysprof-collector.c:
(sysprof_collector_get):
(sysprof_collector_set_counters):
* Source/ThirdParty/libsysprof-capture/sysprof-version-macros.h:
* Source/ThirdParty/libsysprof-capture/sysprof-version.h:

Canonical link: <a href="https://commits.webkit.org/299513@main">https://commits.webkit.org/299513@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/03edb2a7c66cc3771857ae57da609851fdb0da58

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119209 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38891 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29542 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125437 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71272 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39586 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47472 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90558 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59920 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122162 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31561 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106860 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70968 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30605 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24973 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69085 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101007 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25158 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128453 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46116 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34855 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99123 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46483 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103074 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98903 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25153 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44372 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22370 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42695 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45986 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45453 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48801 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47143 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/ThirdParty/libsysprof-capture/README.WebKit:
```diff
@@ -21,8 +21,8 @@ directory, instead of including only the .h.in template. To obtain it,
   meson setup build \
-  	-Dhelp=false -Dsysprofd=none -Dlibsysprof=false \
-	-Dtools=false -Dtests=false -Dexamples=false -Dgtk=false
+    -Dhelp=false -Dsysprofd=none -Dlibsysprof=false \
+    -Dtools=false -Dtests=false -Dexamples=false -Dgtk=false
```
This change seems unnecessary, it's only changing the whitespace.

psaavedra replied:
changes tabs for whitespaces

psaavedra committed:
[GLib] Update the libsysprof-capture source tree to version 48.1
https://bugs.webkit.org/show_bug.cgi?id=298313

Reviewed by Adrian Perez de Castro.

Includes the following upstream changes:

git log --pretty=oneline --abbrev-commit 47.0...48.1 -- src/libsysprof-capture

c7401534 libsysprof-capture: Clean ring buffer memory for counters
b129c7de mapped-ring-buffer: Round to the shared memory boundary on hppa
5151345e libsysprof-capture: fix license of mapped-ring-buffer
45b3f99e libsysprof-capture: unsigned cast before shift
d3058129 libsysprof-capture: Set initial counter id to pid
788b1995 libsysprof-capture: add version macros for 48

* Source/ThirdParty/libsysprof-capture/README.WebKit:
* Source/ThirdParty/libsysprof-capture/mapped-ring-buffer.c:
(mapped_ring_buffer_new_reader):
* Source/ThirdParty/libsysprof-capture/sysprof-collector.c:
(sysprof_collector_get):
(sysprof_collector_set_counters):
* Source/ThirdParty/libsysprof-capture/sysprof-version-macros.h:
* Source/ThirdParty/libsysprof-capture/sysprof-version.h:

Canonical link: https://commits.webkit.org/299513@main
